Guidelet/GuideletLib/Guidelet.py

```diff
@@ -297,18 +306,41 @@ def registerCustomLayouts(self):#common
     self.triple3dCustomLayoutId=506
     layoutLogic.GetLayoutNode().AddLayoutDescription(self.triple3dCustomLayoutId, customLayout)
 
+  # -----------------------------------------------------
+  def onSceneLoaded(self):
+    # Derived classes can override this function
+    pass
 
+  # -----------------------------------------------------
   def setupScene(self):
+    '''
+    ReferenceToRas transform is used in almost all IGT applications. Reference is the coordinate system
+    of a tool fixed to the patient. Tools are tracked relative to Reference, to compensate for patient
+    motion. ReferenceToRas makes sure that everything is displayed in an anatomical coordinate system, i.e.
+    R, A, and S (Right, Anterior, and Superior) directions in Slicer are correct relative to any
+    images or tracked tools displayed.
+    '''
+    self.referenceToRas = slicer.util.getNode('ReferenceToRas')
```

Comment:
If self.referenceToRas exists and it is valid then just use that (don't overwrite it with slicer.util.getNode('ReferenceToRas')  then).